src/enzo/SetEvolveRefineRegion.C

```diff
@@ -23,68 +27,281 @@ int SetEvolveRefineRegion (FLOAT time)
 
   /* Return if not used */
 
-  if (EvolveRefineRegionTime[0] <= FLOAT_UNDEFINED)
+  if (EvolveRefineRegionTime[0] <= FLOAT_UNDEFINED
+      && EvolveMustRefineRegionTime[0] <= FLOAT_UNDEFINED
+      && EvolveCoolingRefineRegionTime[0] <= FLOAT_UNDEFINED)
     return SUCCESS;
 
+  if(debug1){
+    printf("SetEvolveRefineRegion: setting region.\n");
+    printf("SetEvolveRefineRegion: EvolveRefineRegionTime[0]:        %"FSYM"\n",EvolveRefineRegionTime[0]);
+    printf("SetEvolveRefineRegion: EvolveMustRefineRegionTime[0]:    %"FSYM"\n",EvolveMustRefineRegionTime[0]);
+    printf("SetEvolveRefineRegion: EvolveCoolingRefineRegionTime[0]: %"FSYM"\n",EvolveCoolingRefineRegionTime[0]);
+  }
+  
   /* If TimeType is redshift, calculate redshift */
 
   if (ComovingCoordinates) {
     CosmologyComputeExpansionFactor(time, &a, &dadt);
     redshift = (1 + InitialRedshift)/a - 1;
   }
 
-  /* Find closest time step with <time */
-  /* Set time=redshift if that's what we're doing. */
-  if (RefineRegionTimeType == 1) {
-    time = redshift;
-    for(timestep=0; timestep<EvolveRefineRegionNtimes; timestep++){
-      if( time > EvolveRefineRegionTime[timestep] ){
-        break;
+  /* does time evolution for standard refinement regions */
+  if((RefineRegionTimeType == 1) || (RefineRegionTimeType == 0) ){
+  
+    /* Find closest time step with <time */
+    /* Set time=redshift if that's what we're doing. */
+    if (RefineRegionTimeType == 1) {  // redshift
+
+      time = redshift;
```

Comment:
This could be problematic if ComovingCoordinates is False and RefineRegionTimeType == 1. In that case, redshift would not be defined. The user shouldn't be using RefineRegionTimeType == 1 in a non-cosmological simulation but it is possible.

Reply:
Good catch. Maybe just adding an error check in `ReadParameterFile`?

Reply:
Done!